### PR TITLE
[camera_platform_interface] Added stopRecordingVideo

### PR DESCRIPTION
**Pull request by danielroek**
Adds a VideoRecordedEvent to allow the native implementations to finalize the recording of a video. Also adds stopRecordingVideo, which does the same as stopVideoRecording, but gets the result from the stream instead. 

This allows us to implement the maxVideoDuration feature. 

## Pre-launch Checklist

- [X] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [X] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [X] I updated CHANGELOG.md to add a description of the change.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning

### CODE REVIEW

File: packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart [hunk 1]
```diff
@@ -123,7 +128,8 @@ abstract class CameraPlatform extends PlatformInterface {
   /// The length of the recording can be limited by specifying the [maxVideoDuration].
   /// By default no maximum duration is specified,
   /// meaning the recording will continue until manually stopped.
-  /// The video is returned as a [XFile] after calling [stopVideoRecording].
+  /// With [maxVideoDuration] set the video is returned in a [VideoRecordedEvent]
+  /// in the [onVideoRecordedEvent] stream.
```
Comment (mvanbeusekom) [inline packages/camera/camera_platform_interface/lib/src/platform_interface/camera_platform.dart:132]: nit: small update on the description to make it more explanatory.

```suggestion
  /// With [maxVideoDuration] set the video is returned as a [VideoRecordedEvent]
  /// through the [onVideoRecordedEvent] stream when the set duration is reached.
```

**Review by mvanbeusekom — APPROVED** [review]
LGTM, small nit in the description and need to fix formatting for the tree to go green.

File: packages/camera/camera_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ description: A common platform interface for the camera plugin.
 homepage: https://github.com/flutter/plugins/tree/master/packages/camera/camera_platform_interface
 # NOTE: We strongly prefer non-breaking changes, even at the expense of a
 # less-clean API. See https://flutter.dev/go/platform-interface-breaking-changes
-version: 1.5.0
+version: 1.5.1
```
Comment (stuartmorgan) [inline packages/camera/camera_platform_interface/pubspec.yaml:6]: Nit: you are adding functionality, not fixing a bug, so semver calls for incrementing the second number, not the third.

**Comment by stuartmorgan** [comment]
Removing `needs-publishing` as there's an outstanding PR for this package that needs to be landed before it's published again. This fix will be picked up when that is published.

**Pull request merged**